nwbwidgets/base.py

```diff
@@ -371,6 +371,39 @@ def row_to_hover_text(row):
     return "<br>".join(text_rows)
 
 
+class TimeIntervalsSelector(widgets.VBox):
+    InnerWidget = None
+
+    def __init__(self, units, **kwargs):
+        super().__init__()
+        self.units = units
```

Comment:
Can you generalize this to work for any input data, not just units?

Reply:
maybe instead of "units" it could take "input_data" which is a list of objects that go into the inner widget